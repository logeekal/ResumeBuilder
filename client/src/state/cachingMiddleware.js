export const saveState = (state) =>{
    try{
        const serializedState = JSON.stringify(state);
        localStorage.setItem('state',serializedState);
    }catch(err){
        console.error(err)
    }
}


export const loadState = () =>{
    try{
        return undefined
        const serializedState = localStorage.getItem('state');
        if(serializedState === null)
            return undefined;
        console.log('loadState called');
        console.log(serializedState);
        return JSON.parse(serializedState);
    }catch(err){
        console.error(err);
    }
}