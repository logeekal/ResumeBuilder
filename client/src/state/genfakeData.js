import ProfileMeta from '../metadata/ProfileMeta.json';
const shortid = require("shortid"); // shortid.generate() returns a unique "short" id
const txtgen = require("txtgen"); // txtgen.sentence() returns random "readable" sentences
const faker = require("faker"); // faker is used for generating random fake data.
const _ = require("lodash"); // lodash is a utility lib for Javascript

////////////////////////////////////////////////////////
///             Start - Generating personal Infor
////////////////////////////////////////////////////////

function genAddress() {
    return {
        "add1" : {
            "name" :"Address 1",
            "value" : faker.address.streetAddress(false)
        },
        "add2" :{
            "name" : "Address 2",
            "value" : faker.address.secondaryAddress()
        },
        "state" : faker.address.state(),
        "country" : faker.address.country()
    }
}


function genContactInfo(){
    return {
        "phone" : faker.phone.phoneNumber(),
        "email" : faker.internet.email()
    }

}

export  function genPersonalInfo(){
    return {
            "name" : faker.name.firstName() + ' ' + faker.name.lastName(),
            "id" : shortid.generate(),
            "profilePic" : faker.image.avatar(),
            "add1" : faker.address.streetAddress(false),
            "add2" : faker.address.secondaryAddress(),
            "state" : faker.address.state(),
            "country" : faker.address.country(),
            "count"   : 1,
            "visible" : true

        }
}


export function genEducation(){
    var educationMeta = ProfileMeta.education;
    var result ={};
    var details = {};
    var detailObject = {};
    var fieldValList =  {"0":"abc"};
    _.values(educationMeta.children).map((field, index) => {
        if(field.multi){
            detailObject[field.name]= fieldValList;
        }else{
            detailObject[field.name] ="abc";
        }
        
    })
    details[0] = detailObject;
    result.count = 1;
    result.details= details;
    result.visible= true;
    return result;
}

export function genExperience(){
    var expMeta = ProfileMeta.experience;
    var result ={};
    var details = {};
    var detailObject = {};
    var fieldValList = {"0":'abc'};
    _.values(expMeta.children).map((field, index) => {
        if(field.multi){
            detailObject[field.name]= fieldValList;
        }else{
        detailObject[field.name] ="abc";
        }
    })
    details[0] = detailObject;
    result.count = 1;
    result.details= details;
    result.visible= true;
    return result;
}

export  function genSummary(){
    // return {
    //     "summary": txtgen.paragraph(3)
    // }
    return txtgen.paragraph(3);
}

export function genFieldList(){
    return {
        "name" : {
            "editMode" :false
        }
    }
}


export  function genTitle (){
    return  faker.name.jobTitle();
}

export function genLoginInfo(){
    return "SignIn"
}
