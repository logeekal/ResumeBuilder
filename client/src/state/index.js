import {createStore} from 'redux';
import reducers from '../reducers';
import {devToolsEnhancer} from 'redux-devtools-extension'

export const store = createStore(reducers,/* preloadedState, */ devToolsEnhancer(
    // Specify name here, actionsBlacklist, actionsCreators and other options if needed
  ));
//console.log(store.getState());