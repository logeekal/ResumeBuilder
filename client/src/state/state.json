{
  "activeUser": {},
  "loggedUserInfo": {},
  "socialAllowed": {
    "github": { "name": "Github", "url": "./github.png" },
    "stack": { "name": "Stack OverFlow", "url": "./stack.png" },
    "linkedIn": { "name": "LinkedIn", "url": "./linkedin.png" }
  },
  "fields": {
    "personalInfo": {
      "name": {},
      "address": {
        "add1": { "name": "Address 1", "value": "Delhi" },
        "state": {},
        "country": {},
        "add2": { "name": "Address 2", "value": "Delhi" }
      },
      "contact": { "phone": "", "email": {} }
    },
    "socialProfiles": {},
    "contactsInfo": {},
    "title": {},
    "summary": {},
    "skills": [{ "npm": { "rating": "4" } }],
    "allWorkEx": [
      {
        "workEx": {
          "title": {},
          "role": {},
          "company": {},
          "startDate": {},
          "endDate": {},
          "roleDetails": {}
        }
      }
    ],
    "education": {
      "degreeName": {},
      "collegeName": {},
      "country": {},
      "courses": {}
    },
    "Projects": {
      "projectTitle": {},
      "projectDescription": {},
      "whitePaperURL": {}
    }
  },
  "allSkills": {}
}
