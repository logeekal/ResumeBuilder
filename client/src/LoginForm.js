import React from "react";
import './LoginForm.css';
import { store } from './state';
import { updateLoginInfo } from './actions';


  /**
   * handleRegister will action tell state that user wants to SignUp and Not SignIn.
   * This will result is loading of the correct form or switching of the forms.
   * @param {e} e Target for event triggered in Dom.
   */
  const handleRegister = (e) => {
    e.preventDefault();
    store.dispatch(updateLoginInfo('SignUp'));
  }


export const LoginForm = () => {


  return (
    <div className="login">
      <ul className="login_ul flex-container">
        <li>
          <h1 className="login__welcome">Not Yet Registered?</h1>
        </li>
        <li>
          <a className="login__register" href="#" onClick={handleRegister}>Get Started</a>
        </li>
        <li>
          <form>
            <ul className="login__form">
              <li>
                <input
                  className="login__form_name"
                  type="text"
                //   onFocus={this.placeholder.value=""}
                  defaultPlaceholder="Your Email here."
                />
              </li>
              <li>
                <input
                  className="login__form_pass"
                  type="password"
                  placeholder="Your password goes here"
                />
              </li>
              <li>
                <button className="login__form_submit" type="submit" >Login</button>
              </li>
            </ul>
          </form>
        </li>
      </ul>
    </div>
  );
};
