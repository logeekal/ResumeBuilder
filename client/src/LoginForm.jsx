import React from "react";
import "./LoginForm.css";
import { store } from "./state";
import { updateLoginInfo } from "./actions";
import { AUTH_URL } from "./constants";
import {withRouter} from 'react-router-dom';
 import _ from 'lodash';
import { genProfilePayload } from "./utils/AuthReqs";
import { loginInfo } from "./constants/config";


class LoginFormComp extends React.Component {
  constructor(props) {
    super(props);
    this.handleLogin = this.handleLogin.bind(this);
  }


  /**
 * handleRegister will action tell state that user wants to SignUp and Not SignIn.
 * This will result is loading of the correct form or switching of the forms.
 * @param {e} e Target for event triggered in Dom.
 */
 handleRegister = e => {
  e.preventDefault();
  store.dispatch(updateLoginInfo({status :loginInfo.signup, email:  this.refs.email.value }));
};

  async handleLogin(e) {
    e.preventDefault();
    let res;
    let payload = {
      email: this.refs.email.value,
      password: this.refs.password.value,
    };
  
    try {
      res = await fetch(AUTH_URL, {
        method: "POST",
        headers: {
          "Content-Type": "application/json"          
        },
        credentials: "include",
        redirect: "follow",
        referrer : "http://localhost:3000",
        body: JSON.stringify(payload)
      });
      console.log(res);
      // console.log(_.keys(res.headers).toString());
      // res.text().then(result =>{
      //   console.log(result);
      // })

      switch(res.status){
        case 200:
          console.log(`Authentication successfull`);
          store.dispatch(updateLoginInfo({status: "signedIn" , email : this.refs.email.value}));
          this.props.history.push('/profile');
          break;
        case 401:
          console.log(`Invalid Password`);
          break;
        default:
          console.log(`some error in auth`);
          throw res.error;
      }
    }catch (err) {
      console.log(`Some Error occured while authentication: ${err} `);
    }
  }

  render() {
    return (
      <div className="login">
        <ul className="login_ul flex-container">
          <li>
            <h1 className="login__welcome">Not Yet Registered?</h1>
          </li>
          <li>
            <a
              className="login__register"
              id="LoginToRegister"
              href="#"
              onClick={this.handleRegister.bind(this)}
            >
              Get Started
            </a>
          </li>
          <li>
            <form onSubmit={this.handleLogin}>
              <ul className="login__form">
                <li>
                  <input
                    className="login__form_name"
                    type="text"
                    ref="email"
                    //   onFocus={this.placeholder.value=""}
                    // defaultPlaceholder="Your Email here."
                  />
                </li>
                <li>
                  <input
                    className="login__form_pass"
                    type="password"
                    ref="password"
                    placeholder="Your password goes here"
                  />
                </li>
                <li>
                  <button className="login__form_submit" type="submit">
                    Login
                  </button>
                </li>
              </ul>
            </form>
          </li>
        </ul>
      </div>
    );
  }
}


export const LoginForm = withRouter(LoginFormComp);
