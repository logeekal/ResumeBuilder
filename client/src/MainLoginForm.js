import React from 'react';
import { LoginForm } from './LoginForm';
import './MainLoginForm.css';
import { store } from './state';
import { RegisterForm } from './RegisterForm';
import { loginInfo } from './constants/config';
import   RegisterFormRouter from './RegisterForm';



export class MainLoginForm extends React.Component{

    constructor(props){
        super(props);
    }

    getProperForm(){
        const state = store.getState();
        if(store.getState().loginInfo.status === loginInfo.signin){
            return <LoginForm/>
        }else if(store.getState().loginInfo.status == loginInfo.signup){ 
          return  <RegisterFormRouter/>
        } 
    }

    render(){
        return(
        <div className="MainLoginForm">
           {this.getProperForm()}
            
        </div>)
    }
}