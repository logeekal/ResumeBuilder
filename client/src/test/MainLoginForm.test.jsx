//Testing file for LoginForm.text.jsx

import React from 'react';
import renderer from 'react-test-renderer';
import { MainLoginForm } from '../MainLoginForm';
import { store } from '../state';
import { LoginForm } from '../LoginForm';

const delay = (ms: number) =>
  new Promise(resolve => {
    setTimeout(() => {
      resolve();
    }, ms);
  });



test(`Takes the snapshot`,()=>{
    const component =  renderer.create(
        <MainLoginForm/>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    //depends on the state

});

test(`By default MainLogin form should have Login instance and not Register Instance`, ()=>{
    const component =  renderer.create(
        <MainLoginForm/>
    );
    
    const compInstance = component.root;
    const loginInstance =  compInstance.findByProps({id : "LoginToRegister"});

    expect(loginInstance).toEqual(expect.anything());
});


test(`Changes to Signup form when clicked on link with id LoginToRegister link`, async ()=>{
    const component =  renderer.create(
        <MainLoginForm/>
    );

    const event = {
        preventDefault(){},
        target : {'value' : 5}
    }
    
    const compInstance = component.root;
    
    const loginInstance =  compInstance.findByProps({id : "LoginToRegister"});
    console.log(loginInstance);
    const registerInstance = component.getInstance().handleRegister(event);
    //await delay(1);



    //after On Click Events 
    // const registerInstance = compInstance.findByProps({id:  "RegisterToLogin"});
    // expect(registerInstance).anything();

});