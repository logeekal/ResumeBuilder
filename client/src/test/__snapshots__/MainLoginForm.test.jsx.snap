// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Takes the snapshot 1`] = `
<div
  className="MainLoginForm"
>
  <div
    className="login"
  >
    <ul
      className="login_ul flex-container"
    >
      <li>
        <h1
          className="login__welcome"
        >
          Not Yet Registered?
        </h1>
      </li>
      <li>
        <a
          className="login__register"
          href="#"
          id="LoginToRegister"
          onClick={[Function]}
        >
          Get Started
        </a>
      </li>
      <li>
        <form>
          <ul
            className="login__form"
          >
            <li>
              <input
                className="login__form_name"
                type="text"
              />
            </li>
            <li>
              <input
                className="login__form_pass"
                placeholder="Your password goes here"
                type="password"
              />
            </li>
            <li>
              <button
                className="login__form_submit"
                type="submit"
              >
                Login
              </button>
            </li>
          </ul>
        </form>
      </li>
    </ul>
  </div>
</div>
`;
